### PR TITLE
Readability + DRY cleanups from review

### PR DESCRIPTION
**Pull request by jhodapp**
## Description
Post-review readability and DRY cleanups for the coaching-session title/topics/views work that recently merged to `main`. Pure refactor, no behavior change. Surfaced by a senior-readability review pass; split out into its own PR rather than amending the already-merged feature PRs.

### Changes
* `entity_api::normalize_title_in_update_map`: collapse the nested-`Option` sentinel + `match`/`if let` two-step into one `if let` with `bool::then`.
* `entity_api::coaching_session_topic::update`: use `topic.into()` + targeted `Set` like every sibling, instead of enumerating all 13 `ActiveModel` fields.
* Extractors: factor the duplicated `Path`-parse block and the repeated `(NOT_FOUND, "NOT FOUND")` literal into shared `parse_path_id_from_parts()` and `not_found()` helpers.
* `params::coaching_session`: de-dup the clearable-title update-map insert into one `insert_title_update()` helper shared by `UpdateParams` and `TitleUpdateParams`.
* Move the topic-access extractor tests into a separate `coaching_session_topic_access_tests.rs` (frozen-test convention; matches the sibling params module).
* Title controller test: `result.expect(...)` instead of `if let Err(e) { panic!(...) }`.

### Testing Strategy
No behavior change, so existing tests are the contract. Verified locally on this branch (off latest `main`):
* `cargo test -p entity_api -p domain -p web --features "domain/mock,web/mock"` — green (183 / 219 / 102 + gated suites, 0 failures), including all 12 moved topic-access extractor tests and the `normalize_*` edge-case tests.
* `cargo fmt --check` and `cargo clippy` clean.

### Concerns
* The large line counts are the test-file extraction (~681 lines moved out of the extractor, ~631 into the new `_tests.rs`); the net production change is small.
* No migration, schema, or wire-contract changes.

### CODE REVIEW

File: entity_api/src/coaching_session_topic.rs [hunk 1]
```diff
@@ -81,22 +81,10 @@ pub async fn find_including_deleted_by_id(
...
-        // A body edit is a deliberate re-commit, so it settles the undo window even on a
-        // deferred/moved topic: a subsequent undo then returns 422 rather than reverting the edit.
-        undo_snapshot: Set(None),
-        deleted_at: Unchanged(topic.deleted_at),
-        created_at: Unchanged(topic.created_at),
-        updated_at: Set(chrono::Utc::now().into()),
-    };
+    let mut active: ActiveModel = topic.into();
+    active.body = Set(body);
+    active.undo_snapshot = Set(None);
```
[the remark this answers is not in the document]
Comment (jhodapp) [reply entity_api/src/coaching_session_topic.rs]: Good catch on the asymmetry. Restored a terse intent comment matching the `set_priority`/`set_status` siblings: `// a body edit settles the undo window` (in `c29b265c`). Kept it to one line rather than the original multi-line block, per our comment-terseness preference.
